gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/NonAbstractGradleType.java

```diff
@@ -53,18 +58,54 @@ public final class NonAbstractGradleType extends GradleGuideBugChecker
             .named("create");
     private static final Supplier<Type> CLASS_TYPE_SUPPLIER = Suppliers.typeFromString("java.lang.Class");
 
+    private static final Set<String> SUPPORTED_PROPERTY_TYPES = Set.of(
+            "org.gradle.api.provider.Property",
+            "org.gradle.api.provider.ListProperty",
+            "org.gradle.api.provider.SetProperty",
+            "org.gradle.api.provider.MapProperty",
+            "org.gradle.api.file.RegularFileProperty",
+            "org.gradle.api.file.DirectoryProperty",
+            "org.gradle.api.provider.Provider",
+            "org.gradle.api.model.ObjectFactory");
+
     @Override
     public Description matchClass(ClassTree tree, VisitorState state) {
         if (tree.getKind().equals(Kind.INTERFACE)) {
             return Description.NO_MATCH;
         }
 
-        if (tree.getModifiers().getFlags().contains(Modifier.ABSTRACT)) {
-            return Description.NO_MATCH;
-        }
-
         if (IS_TASK.matches(tree, state)) {
-            return buildDescription(tree).build();
+            if (!tree.getModifiers().getFlags().contains(Modifier.ABSTRACT)) {
+                return buildDescription(tree).build();
+            }
+            return tree.getMembers().stream()
+                    .filter(member -> member instanceof MethodTree)
+                    .map(member -> (MethodTree) member)
+                    .filter(method -> isManagedPropertyGetter(method, state))
+                    .map(method -> {
+                        if (!method.getModifiers().getFlags().contains(Modifier.ABSTRACT)) {
+                            return buildDescription(method)
+                                    .setMessage("Gradle managed property getter methods must be abstract. "
```

Comment:
I would not assume people know what a "Gradle managed property" is and instead say "properties on Tasks should be declared abstract." followed by the rest of the error message.